s3/src/main/scala/akka/stream/alpakka/s3/S3Settings.scala

```diff
@@ -51,47 +54,78 @@ object S3Settings {
    */
   def apply()(implicit system: ActorSystem): S3Settings = apply(system.settings.config)
 
+  /**
+   * Scala API: Creates [[S3Settings]] from the [[Config]] attached to an [[ActorSystem]].
+   */
+  def apply(configurationPrefix: String)(implicit system: ActorSystem): S3Settings =
+    apply(system.settings.config, configurationPrefix)
+
   /**
    * Scala API: Creates [[S3Settings]] from a [[Config]] object.
    */
-  def apply(config: Config): S3Settings = {
-    val bufferType = config.getString("akka.stream.alpakka.s3.buffer") match {
+  def apply(config: Config, configurationPrefix: String = "akka.stream.alpakka.s3"): S3Settings = {
```

Comment:
@ennru added `configurationPrefix` for the discussed configuration purpose

Reply:
Therefore pr affects https://github.com/akka/alpakka/issues/388 and partially https://github.com/akka/alpakka/issues/606

Reply:
Great. 
You can make better use of Config by using `val config = configIn.getConfig(configurationPrefix)` and removing the prefix down the road.

Reply:
@ennru updated the method and added a new method to simplify creation of S3Client with custom S3Settings https://github.com/akka/alpakka/pull/645/files#diff-edcf65bc8d22d487709bda00666179c1R83